src/backend/gporca/libgpopt/src/translate/CTranslatorExprToDXL.cpp

```diff
@@ -449,6 +449,22 @@ CTranslatorExprToDXL::CreateDXLNode(CExpression *pexpr,
 
 		return dxlnode;
 	}
+	if (COperator::EopPhysicalMotionGather == ulOpId ||
+		COperator::EopPhysicalMotionBroadcast == ulOpId ||
+		COperator::EopPhysicalMotionHashDistribute == ulOpId ||
+		COperator::EopPhysicalMotionRoutedDistribute == ulOpId ||
+		COperator::EopPhysicalMotionRandom == ulOpId)
+	{
+		gpos::IntPtrArray *inputSegmentInfo = GetInputSegIdsArray(pexpr);
+
+		if (NULL != inputSegmentInfo)
+		{
+			pexpr->SetMotionInputSegmentsNumber(inputSegmentInfo->Size());
```

Comment:
I think, this should be placed in function begin to cover two cases of return below.

Reply:
If you mean two cases above then I don't think it's necessary because our case works with motions only and those two work with table scan and external scan

Reply:
Done